crud.py

```diff
@@ -0,0 +1,44 @@
+from sqlalchemy.orm import Session
+
+import models, schemas
+
+
+def get_all_authors(db_session: Session):
+    authors = db_session.query(models.Author).all()
+    return authors
+
+
+def get_author_by_name(db_session: Session, name: str):
+    return db_session.query(models.Author).filter(models.Author.name == name).first()
+
+
+def get_author_by_id(db_session: Session, _id: int):
+    return db_session.query(models.Author).filter(models.Author.id == _id).first()
+
+
+def create_author(db_session: Session, author: schemas.AuthorCreate):
+    db_new_author = models.Author(name=author.name, bio=author.bio)
+    db_session.add(db_new_author)
+    db_session.commit()
+    db_session.refresh(db_new_author)
+    return db_new_author
+
+
+def get_all_books(db_session: Session, author_id: int | None = None):
+    queryset = db_session.query(models.Book)
+    if author_id:
+        queryset.filter(models.Book.author_id == author_id)
```

Comment:
The filter condition is not applied to the query. You need to assign the result of `queryset.filter(...)` back to `queryset` to apply the filter correctly.